app/lib/search/backend.dart

```diff
@@ -138,6 +147,71 @@ class SearchBackend {
   }
 }
 
+class PopularityStorage {
+  final Storage storage;
+  final Bucket bucket;
+
+  final Map<String, double> _values = {};
+
+  PopularityStorage(this.storage, this.bucket);
+
+  double operator [](String package) => _values[package];
```

Comment:
I would prefer not overloading operators, to avoid confusion on call sites if this is a map or something else. Maybe rename this to `lookup`?